odoo/fields.py

```diff
@@ -3047,6 +3058,518 @@ def _record_ids_per_res_model(self, records):
+
+        return values_list
+
+    @classmethod
+    def _list_to_dict(cls, values_list):
+        """Convert a list of properties with definition into a dict {name: value}.
+
+        To not repeat data in database, we only store the value of each property on
+        the child. The properties definition is stored on the parent.
+
+        :param values_list: List of properties definition and value
+        :return: Generate a dict {name: value} from this definitions / values list
+        """
+        assert is_list_of(values_list, dict)
+
+        dict_value = {}
+        for property_definition in values_list:
+            property_value = property_definition.get('value')
+            property_type = property_definition.get('type')
+            property_model = property_definition.get('model')
+
+            if property_type in ('many2one', 'many2many') and property_model and property_value:
+                # many2one: (1337, "Mitchel Admin")
+                # keep only the record id in the dict that will be stored in database
+                # many2many: [(1336, "Marc Demo"), (1337, "Mitchel Admin")]
+                # [(1, 2, 3), 'res.partner']
+                property_value = [property_value, property_model]
+
+            dict_value[property_definition['id']] = property_value
+
+        return dict_value
+
+    @classmethod
+    def _dict_to_list(cls, values_dict, properties_definition):
+        """Convert a dict of {property: value} into a list of property definition with values.
+
+        :param values_dict: JSON value coming from the child table
+        :param properties_definition: Properties definition coming from the parent table
+        :return: Merge both value into a list of properties with value
+            Ignore every values in the child that is not defined on the parent.
+        """
+        assert is_list_of(properties_definition, dict)
+        values_list = copy.deepcopy(properties_definition)
+
+        for property_definition in values_list:
+            property_definition['value'] = values_dict.get(property_definition['id'])
+
+        return values_list
+
+    def _add_default_values(self, env, values):
+        """Read the properties definition to add default values.
+
+        Default values are defined on the parent, in the 'default' key.
+
+        :param env: environment
+        :param values: All values that will be written on the record
+        :return: Return the default values in the "dict" format
+        """
+        properties_values = values.get(self.name) or {}
+
+        if isinstance(properties_values, list):
+            properties_values = self._list_remove_display_name(properties_values)
+            properties_values = self._list_to_dict(properties_values)
+
+        if not values.get(self.parent_record):
+            # parent is not given in the value, can not find properties definition
+            return properties_values
+
+        parent_id = values[self.parent_record]
```

Comment:
Does it work if the parent itself is given in the default through context usage? Like 'default_parent_id: 3'?